apps/src/templates/projects/projectsRedux.js

```diff
@@ -378,22 +385,23 @@ const updateProjectNameOnServer = (project) => {
       contentType: 'application/json;charset=UTF-8',
       data: JSON.stringify(project)
     }).done((data) => {
-      dispatch(saveSuccess(project.id));
+      dispatch(saveSuccess(project.id, lastUpdatedAt));
```

Comment:
this request to update the project name should be returning a server timestamp, which you can pull out of `data` here. the server timestamp is what you should be using to update the timestamp associated with the project. otherwise, we have a mix of server and client timestamps, which can lead to weird locale and timezone bugs.

You would pass that server timestamp to `saveSuccess` here. Then, in the place that the existing reducer code updates the renamed project, you would just also set the new timestamp:
https://github.com/code-dot-org/code-dot-org/blob/af0543ef4a7242d19ae3639856f390afadde3f5e/apps/src/templates/projects/projectsRedux.js#L292-L297

You can then remove the generation of the client-side timestamp and all the places you pass it around to.